src/pages/nfts/[...address].tsx

```diff
@@ -0,0 +1,378 @@
+import { NextPage } from "next"
+import { AppProps } from 'next/app'
+import { gql } from '@apollo/client'
+import { isNil } from 'ramda'
+import client from '../../client'
+import { Link } from 'react-router-dom';
+import {
+  WalletMultiButton,
+} from '@solana/wallet-adapter-react-ui'
+import NextLink from 'next/link'
+import { Route, Routes } from 'react-router-dom'
+import Offer from '../../components/Offer';
+import SellNft from '../../components/SellNft';
+
+const solSymbol = '◎'
+const SUBDOMAIN = process.env.MARKETPLACE_SUBDOMAIN
+
+export async function getServerSideProps(ctx: any) {
+  const {
+    data: { storefront, nft },
+  } = await client.query<GetNftPage>({
+    query: gql`
+      query GetNftPage($subdomain: String!, $address: String!) {
+        storefront(subdomain: $subdomain) {
+          title
+          description
+          logoUrl
+          faviconUrl
+          bannerUrl
+          ownerAddress
+        }
+        nft(address: $address) {
+          name
+          address
+          image
+          sellerFeeBasisPoints
+          mintAddress
+          description
+          attributes {
+            traitType
+            value
+          }
+        }
+      }
+    `,
+    variables: {
+      subdomain: SUBDOMAIN,
+      address: ctx.params.address[0],
```

Comment:
In order to get next serving the nest react router paths I had to convert to fuzzy param list that splits on path so the previous named address param is always in the first slot. react-router picks up the rest to do the actions area.